app/models/admin_only/app_configuration.rb

```diff
@@ -28,25 +61,51 @@ class AppConfiguration < ApplicationRecord
                       styles: { standard: ['234x39#'] },
                       default_style: :standard
 
+    # The site meta image (used in OpenGraph meta info, etc.).
+    # This must be shown via a public URL to (e.g. for Facebook). If it were
+    # simply served up as an asset then we would manually have to apply the fingerprinting
+    # as Sockets does.
+    # This image _must be_ in the database; that's why there is no default for it
+    # and  :validate_attachment_presence is used
+    has_attached_file :site_meta_image,
+                      url: :url_for_images
+
+
+    validates_presence_of :site_name, :site_meta_title
+
+    validates_attachment_presence :site_meta_image
+
+
     validates_attachment_content_type :chair_signature, :shf_logo,
                                       :h_brand_logo, :sweden_dog_trainers,
-                                      content_type:  /\Aimage\/.*(jpeg|png)\z/
+                                      :site_meta_image,
+                                      content_type: /\Aimage\/.*(jpeg|png)\z/
 
     validates_attachment_file_name :chair_signature, :shf_logo,
                                    :h_brand_logo, :sweden_dog_trainers,
+                                   :site_meta_image,
                                    matches: [/png\z/, /jpe?g\z/]
 
 
-    scope :config_to_use, -> { last }
+
+    class << self
+
+      def instance
+        first_or_create!(singleton_guard: 0)
+      end
```

Comment:
Nice!  

I'm curious as to why these class methods had to be added by reopening the class (within the class definition)?

Reply:
Need to use the class << method style so that I could use `alias_method`: 
   
   `alias_method` only works on instance methods.  Using this style opens up the eigenclass (== the "class of this class") and the methods are the _instance methods_ of the eigenclass.

I added a small comment about this.  (Since you had this question, others definitely will, too.) :-)

Since the following class method does not have that requirement, I changed it to use the standard `def self.` style.